tests/support/__init__.py

```diff
@@ -340,12 +362,17 @@ def temppath(relative_path, test_case, ensure_dir=False, skip_clean=False):
     """Get absolute temp path for relative_path"""
     assert isinstance(test_case, MigTestCase)
     tmp_path = os.path.join(TEST_OUTPUT_DIR, relative_path)
+    return _temppath(tmp_path, test_case, ensure_dir=ensure_dir, skip_clean=skip_clean)
+
+
+def _temppath(tmp_path, test_case, ensure_dir=False, skip_clean=False):
```

Comment:
Sorry if this comes out blunt after spending a significant amount of time figuring this code bit out, but on a closer look I don't like this `temppath` + `_temppath` construct and the out-of-module use of the latter at all. Probably also because I don't really see the point of exposing this part of the existing `temppath` as a new semi-private function with a very similar name and then using it directly in test modules.
If the `TEST_OUTPUT_DIR` anchoring should be skipped in some use cases there, that could easily be achieved with, say, a `skip_output_anchor` argument instead. That would keep all external use to just `temppath` and avoid the naming and functionality confusion as well.
I'll see if I can figure that out during merge now that I'm already halfway through.